.githooks/pre-commit

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+changed_files = `git status --porcelain`.split(/\n/)
+unstaged_files = `git ls-files -m`.split(/\n/)
+
+changed_files = changed_files.map do |file|
+  /^\s*[AM]+\s+(?<filename>.+)$/ =~ file # Find added or removed files
+  filename
+end.compact
+
+changed_files -= (unstaged_files - changed_files)
```

Comment:
Is this right? Does it not evaluate to `changed_files = 2 * changed_files - unstaged_files`?

Reply:
I did think this line looked dodgy.  I've modified the script to have better variable names as well.